File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.ArrayList;
import java.util.LinkedList;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;

        ArrayList<Character> list = new ArrayList<>();
        char find = (char) (location + 65);

        LinkedList<Character> wordQ = new LinkedList<Character>();
        LinkedList<Integer> priorQ = new LinkedList<>();
        for (int i = 0; i < priorities.length; i++) {
            char tmp = (char) (i + 65);
            wordQ.add(tmp);
            priorQ.add(priorities[i]);
        }

        while (!wordQ.isEmpty()) {
            int max = 0;
            for(int i = 0; i < priorQ.size(); i++) {
                max = Math.max(max, priorQ.get(i));
            }

            char curr = wordQ.peek();
            int currP = priorQ.peek();

            if(priorQ.peek() == max) {
                list.add(wordQ.poll());
                priorQ.poll();
            }
            else {
                wordQ.poll();
                wordQ.add(curr);
                priorQ.poll();
                priorQ.add(currP);
            }
        }


        answer = list.indexOf(find) + 1;

        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0

    dict_priorities = {chr(65 + x) : priorities[x] for x in range(0, len(priorities))}    

    queue = deque(dict_priorities.items())

    while queue:
        key, value = queue.popleft()

        if queue and max([item[1] for item in queue]) > value:
            queue.append((key, value))
        else:
            answer += 1

            if ord(key) == 65 + location:
                break

    return answer